src/wasm2js.h

```diff
@@ -173,8 +173,12 @@ class Wasm2JSBuilder {
     // and store would need to do a check. Given that, we can just ignore
     // implicit traps like those when optimizing. (When not optimizing, it's
     // nice to see codegen that matches wasm more precisely.)
+    // Similarly, when targeting JS, it is important to clarify this for each pass,
+    // so no optimizations past level 0 end up adding non-JS things that cannot be
+    // targeted in the final output.
```

Comment:
```suggestion
    // It is also important to prevent the optimizer from adding new things that
    // require additional lowering, as we could hit a cycle.
```